spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala

```diff
@@ -34,7 +34,9 @@ class CassandraJoinRDD[L, R] private[connector](
     val where: CqlWhereClause = CqlWhereClause.empty,
     val limit: Option[Long] = None,
     val clusteringOrder: Option[ClusteringOrder] = None,
-    val readConf: ReadConf = ReadConf())(
+    val readConf: ReadConf = ReadConf(),
+    manualRowReader: Option[RowReader[R]] = None,
+    manualRowWriter: Option[RowWriter[L]] = None)(
```

Comment:
I can see two usages:
- https://github.com/datastax/spark-cassandra-connector/blob/SPARKC-202/spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/RDDFunctions.scala#L142-L142
- https://github.com/datastax/spark-cassandra-connector/blob/SPARKC-202/spark-cassandra-connector-java/src/main/java/com/datastax/spark/connector/japi/RDDJavaFunctions.java#L108-L108

Only one of these usages passes a rowWriter/rowReader and another one passes a factory. Any reason why the Java one needed this refactoring while the Scala one is ok without it? Or is it an omission? 

Anyway, I'd pass a rowWriter/rowReader to any RDD implementations, as it is much safer for its serializability (no need to do @transient). Wdyt?